Symfony/CS/Fixer/Contrib/MethodArgumentDefaultValueFixer.php

```diff
@@ -0,0 +1,236 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'In method arguments there must not be arguments with default values before non-default ones.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+        for ($i = 0, $l = $tokens->count(); $i < $l; ++$i) {
+            if ($tokens[$i]->isGivenKind(T_FUNCTION)) {
+                $this->fixFunctionDefinition($tokens, $i);
+            }
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     */
+    private function fixFunctionDefinition(Tokens $tokens, $index)
+    {
+        $examinedIndex = $tokens->getNextTokenOfKind($index, $this->argumentBoundaryTokens);
+        $lastNonDefaultArgumentIndex = $this->getLastNonDefaultArgumentIndex($tokens, $index);
+
+        while (
+            $examinedIndex < $lastNonDefaultArgumentIndex &&
+            $this->hasDefaultValueAfterIndex($tokens, $examinedIndex)
+        ) {
+            $nextRelevantIndex = $tokens->getNextTokenOfKind($examinedIndex, $this->variableOrTerminatorTokens);
+
+            if (!$tokens[$nextRelevantIndex]->isGivenKind(T_VARIABLE)) {
+                break;
+            }
+
+            if (
+                $this->isDefaultArgumentAfterIndex($tokens, $nextRelevantIndex - 1) &&
+                $nextRelevantIndex - 1 < $lastNonDefaultArgumentIndex
+            ) {
+                $this->removeDefaultArgument($tokens, $nextRelevantIndex);
+            }
+            $examinedIndex = $nextRelevantIndex;
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return int|null
+     */
+    private function getLastNonDefaultArgumentIndex(Tokens $tokens, $index)
+    {
+        $nextRelevantTokenIndex = $tokens->getNextTokenOfKind($index, $this->variableOrTerminatorTokens);
+
+        if (null === $nextRelevantTokenIndex) {
+            return;
+        }
+
+        $lastNonDefaultArgumentIndex = null;
+
+        while ($tokens[$nextRelevantTokenIndex]->isGivenKind(T_VARIABLE)) {
+            if (!$tokens[$tokens->getNextMeaningfulToken($nextRelevantTokenIndex)]->equals('=') &&
+                !$this->isEllipsis($tokens, $nextRelevantTokenIndex)
+            ) {
+                $lastNonDefaultArgumentIndex = $nextRelevantTokenIndex;
+            }
+
+            $nextRelevantTokenIndex = $tokens->getNextTokenOfKind($nextRelevantTokenIndex, $this->variableOrTerminatorTokens);
+        }
+
+        return $lastNonDefaultArgumentIndex;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $variableIndex
+     *
+     * @return bool
+     */
+    private function isEllipsis(Tokens $tokens, $variableIndex)
+    {
+        if (PHP_VERSION_ID < 50600) {
+            return false;
```

Comment:
Exactly. When I try to run the fixer on input code with ... using 5.5 interpreter, fixer should not remove the default value. We need to make prevention for that to happen. Because not all users use linting